File: src/main/java/it/unibo/crabinv/Controller/entities/bullets/PlayerBulletController.java
package it.unibo.crabinv.Controller.entities.bullets;

import it.unibo.crabinv.Controller.entities.entity.EntityAbstractController;
import it.unibo.crabinv.Model.entities.bullets.BulletPlayer;
import it.unibo.crabinv.Model.entities.entity.Delta;

/**
 * It's the PlayerBulletController that handles the PlayerBullets.
 */
public class PlayerBulletController extends EntityAbstractController<BulletPlayer> implements BulletController {

    /**
     * It's the constructor of the EnemyBulletController.
     *
     * @param entity it's the entity needed to modify it
     */
    public PlayerBulletController(final BulletPlayer entity) {
        super(entity);
    }

    @Override
    public void update(final Delta delta) {
        move(delta);
    }

    /**
     * Gives the speed of the bullet.
     *
     * @return the speed of the bullet
     */
    public double getSpeed() {
        return super.getEntity().getSpeed();
    }

    /**
     * Tells the bullet to go to a specific direction for 1 tick.
     *
     * @param delta either -1, 0 or 1, the former moves to the left, the latter moves to the right
     */
    private void move(final Delta delta) {
        super.getEntity().move(delta);
    }

}

File: src/main/java/it/unibo/crabinv/controller/entities/bullets/PlayerBulletController.kt
package it.unibo.crabinv.controller.entities.bullets

import it.unibo.crabinv.controller.entities.entity.AbstractEntityController
import it.unibo.crabinv.model.entities.bullets.BulletPlayer
import it.unibo.crabinv.model.entities.entity.Delta

/**
 * It's the PlayerBulletController that handles the PlayerBullets.
 * @param entity it's the entity needed to modify it
 */
class PlayerBulletController(
    entity: BulletPlayer?,
) : AbstractEntityController<BulletPlayer?>(entity),
    BulletController {
    override fun update(delta: Delta) {
        move(delta)
    }

    val speed: Double
        /**
         * Gives the speed of the bullet.
         * @return the speed of the bullet
         */
        get() = super.entity!!.speed

    /**
     * Tells the bullet to go to a specific direction for 1 tick.
     * @param delta either -1, 0 or 1, the former moves to the left, the latter moves to the right
     */
    private fun move(delta: Delta) {
        super.entity!!.move(delta)
    }
}
